src/tuned_models.jl

```diff
@@ -278,27 +291,134 @@ function event(metamodel,
-    pmap(metamodels) do m
-        event(m, resampling_machine, verbosity, tuning, history, state)
+  resampling_machine = resampling_machines[1]
+
+  ret = if verbosity < 1
+       pmap(metamodels) do m
+            event(m, resampling_machine, verbosity, tuning, history, state)
+       end
+  else
+      n_metamodels = length(metamodels)
+      p = Progress(n_metamodels,
+                 dt = 0,
+                 desc = "Evaluating over $(n_metamodels) metamodels: ",
+                 barglyphs = BarGlyphs("[=> ]"),
+                 barlen = 25,
+                 color = :yellow)
+      update!(p,0)
+      progress_pmap(metamodels, progress=p) do m
+            event(m, resampling_machine, verbosity, tuning, history, state)
+      end
+
+  end
+    
+    return ret
+end
+
+@static if VERSION >= v"1.3.0-DEV.573"
+# one machine for each thread; cycle through available threads:
+function assemble_events(metamodels,
+                         resampling_machines,
+                         verbosity,
+                         tuning,
+                         history,
+                         state,
+                         acceleration::CPUThreads)
+    
+    if Threads.nthreads() == 1
+        return assemble_events(metamodels,
+                         resampling_machines,
+                         verbosity,
+                         tuning,
+                         history,
+                         state,
+                         CPU1())
+   end
+    n_metamodels = length(metamodels)
+    n_threads = Threads.nthreads()
+    
+    
+    ret = Array{Any, 1}(undef, n_metamodels)
+    verbosity < 1 || (p = Progress(n_metamodels,
+                 dt = 0,
+                 desc = "Evaluating over $(n_metamodels) metamodels: ",
+                 barglyphs = BarGlyphs("[=> ]"),
+                 barlen = 25,
+                 color = :yellow))
+    verbosity < 1 || update!(p,0)
+    lock_ = ReentrantLock()
+    
+    @sync for parts in Iterators.partition(1:n_metamodels, max(1,floor(Int, n_metamodels/n_threads)))   
+    
+    Threads.@spawn begin        
+        foreach(parts) do m
+            id = Threads.threadid()
+            if !haskey(resampling_machines, id)
+               resampling_machines[id] =
+                   machine(Resampler(model= resampling_machines[1].model.model,
+                      resampling    = resampling_machines[1].model.resampling,
+                      measure       = resampling_machines[1].model.measure,
+                      weights       = resampling_machines[1].model.weights,
+                      operation     = resampling_machines[1].model.operation,
+                      check_measure = resampling_machines[1].model.check_measure,
+                      repeats       = resampling_machines[1].model.repeats,
+                      acceleration  = resampling_machines[1].model.acceleration),
+                      resampling_machines[1].args...)
+            end
```

Comment:
Nice. You avoid deep copy here. I had to think very carefully about the `model.model` but I see that this get's *replaced* and not just mutated in the `event` call below.